src/Common/src/Interop/Unix/libc/Interop.strerror.cs

```diff
@@ -10,15 +10,71 @@ internal static partial class Interop
 {
     internal static partial class libc
     {
-        internal unsafe static string strerror(int errno) // thread-safe version of strerror that internally uses strerror_r, which is thread-safe
+        internal static string strerror(int errno)
         {
-            const int bufferLength = 1024; // length long enough for most any Unix error messages
-            byte* buffer = stackalloc byte[bufferLength];
-            IntPtr errorPtr = (IntPtr)strerror_r(errno, buffer, (IntPtr)bufferLength);
-            return Marshal.PtrToStringAnsi(errorPtr); // TODO: Use the proper Encoding
+            string result = s_isGnu.Value ? strerror_gnu(errno) : strerror_xsi(errno);
+            // System.Diagnostics.Debug.WriteLine("strerror: " + result + "\n" + System.Environment.StackTrace); // uncomment to aid debugging
+            return result;
         }
 
+        // strerror is not thread-safe; instead, there is the strerror_r function, which is thread-safe.
+        // However, there are two versions of strerror_r:
+        // - GNU: char* strerror_r(int, char*, size_t);
+        // - XSI: int   strerror_r(int, char*, size_T);
+        // The former may or may not use the supplied buffer, and returns the error message string.
+        // The latter stores the error message string into the supplied buffer.
+        // Due to the different return values, we don't want to just choose one signature
+        // and try to use a heuristic to deduce which version we're dealing with, as that
+        // could result in a stack imbalance due to the varied return result size on 64-bit.  Instead,
```

Comment:
It is a good idea to have two different signatures, but the mismatch would not result in stack imbalance in this case - I would delete part of the comment that mentions it. Both char* and int has been always returned in register on all known platforms.